File: services/iposCron/model/applications/patent/ApplicationSummary.go
package patent

type ApplicationSummary struct {
	ApplicationNum                        string `json:"applicationNum" bson:"applicationNum"`
	ApplicationType                       string `json:"applicationType" bson:"applicationType"`
	ApplicationStatus                     string `json:"applicationStatus" bson:"applicationStatus"`
	PublicationPatentNumForOldApplication string `json:"PublicationPatentNumForOldApplication" bson:"PublicationPatentNumForOldApplication"`
	TitleOfInvention                      string `json:"TitleOfInvention" bson:"TitleOfInvention"`
	FilingDate                            string `json:"filingDate" bson:"filingDate"`
	LodgementDate                         string `json:"lodgementDate" bson:"lodgementDate"`
	DateOfPublication                     string `json:"dateOfPublication" bson:"dateOfPublication"`
	IPC                                   string `json:"ipc" bson:"ipc"`
}

File: services/ipos/src/main/java/com/cs5224/ipos/model/documents/ApplicationSummary.java
package com.cs5224.ipos.model.documents;

import lombok.Data;
import org.springframework.data.mongodb.core.mapping.Field;

@Data
public class ApplicationSummary {

    @Field("applicationNum")
    String applicationNum;

    @Field("applicationType")
    String applicationType;

    @Field("applicationStatus")
    String applicationStatus;

    @Field("PublicationPatentNumForOldApplication")
    String publicationPatentNumForOldApplication;

    @Field("TitleOfInvention")
    String titleOfInvention;

    @Field("filingDate")
    String filingDate;

    @Field("lodgementDate")
    String lodgementDate;

    @Field("dateOfPublication")
    String dateOfPublication;

    @Field("ipc")
    String ipc;

}
